app/models/ems_refresh/save_inventory_helper.rb

```diff
@@ -50,10 +50,13 @@ def save_inventory_multi(association, hashes, deletes, find_key, child_keys = []
     remove_keys = Array.wrap(extra_keys) + child_keys
 
     record_index = TypedIndex.new(association, find_key)
+    reconnect_index = TypedIndex.new(reconnect_from, find_key)
```

Comment:
hm, this like can add another huge memory peak, if we collect enough disconnected_records